LLama.Unittest/LLamaEmbedderTests.cs

```diff
@@ -42,37 +42,42 @@ private async Task CompareEmbeddings(string modelPath)
         var spoon = (await embedder.GetEmbeddings("The spoon is not real")).Single().EuclideanNormalization();
         Assert.DoesNotContain(float.NaN, spoon);
 
-        var generator = (IEmbeddingGenerator<string, Embedding<float>>)embedder;
-        Assert.NotNull(generator.GetService<EmbeddingGeneratorMetadata>());
-        Assert.Equal(nameof(LLamaEmbedder), generator.GetService<EmbeddingGeneratorMetadata>()?.ProviderName);
-        Assert.NotNull(generator.GetService<EmbeddingGeneratorMetadata>()?.DefaultModelId);
-        Assert.NotEmpty(generator.GetService<EmbeddingGeneratorMetadata>()?.DefaultModelId!);
-        Assert.Same(embedder, generator.GetService<LLamaEmbedder>());
-        Assert.Same(generator, generator.GetService<IEmbeddingGenerator<string, Embedding<float>>>());
-        Assert.Null(generator.GetService<string>());
-
-        var embeddings = await generator.GenerateAsync(
-        [
-            "The cat is cute",
+        if (false)
```

Comment:
Does this need resolving before merge?

Reply:
`generator.GetService<EmbeddingGeneratorMetadata>()` uses the context and thus will fail because for a context efficient handling we do not keep the context. This was the main aim of this PR.  

The code in the test assumes that there is a context. I think that for the test code to work we would need some extra work to create an embedding service that keeps the context (this could be done in a next PR, if anybody is interested in to do it). The aim of the embedder in our code is different. My opinion is that the test code is wrong because it assume that the embedder is a live service, and it should not be for efficiently handling of GPU memory. There are two options, delete the test code or leave it in switched off with the TODO comment I have added.